federation/pkg/crud/secret.go

```diff
@@ -0,0 +1,93 @@
+/*
```

Comment:
The goal of this refactor commit is separating the integration-specific stuff from the stuff that can be reused in integration/e2e/controllers.  This is the stuff that can be reused.